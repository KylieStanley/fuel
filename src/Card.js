import React from 'react'
import './styles/main.scss';

const Card = () => {
	return (
		<div className="card">
		</div>
		
	)
}

export default Card