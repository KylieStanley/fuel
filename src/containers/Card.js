import React, { Component } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { Link, withRouter } from 'react-router-dom'
import { addFavorite, removeFavorite, selectCard } from '../actions'
import '../styles/main.scss'


export class Card extends Component {
	toggleFavorite = () => {
		const {
			favorites, recipe, removeFavorite, addFavorite
		} = this.props

		const match = favorites.find(favorite => favorite.url === recipe.url)

		if (match) {
			removeFavorite(recipe)
		} else {
			addFavorite(recipe)
		}
	}

	render() {
		const { favorites, recipe, selectCard } = this.props
		const favorite = favorites.find(favorite => favorite.url === recipe.url)
		const favoriteClass = favorite ? <i className="favorite fas fa-heart" /> : <i className="far fa-heart" />

		return (
			<div className="card">
				<div className="image" style={{ backgroundImage: `url(${recipe.image})` }} />
				<div className="recipe-info">
					<Link to={`${this.props.match.url}/modal`} onClick={selectCard.bind(null, recipe)}>
						<h3 className="recipe-name">{recipe.name}</h3>
					</Link>
					<div className="card-bottom">
						<button type="button" className="fave-btn" onClick={this.toggleFavorite}>
							{favoriteClass}
						</button>
						{recipe.dietLabel.map(label => <p>{label}</p>)}
					</div>
				</div>
			</div>
		)
	}
}

export const mapStateToProps = (state) => ({
	favorites: state.favorites
})

export const mapDispatchToProps = (dispatch) => ({
	addFavorite: (recipe) => dispatch(addFavorite(recipe)),
	removeFavorite: (recipe) => dispatch(removeFavorite(recipe)),
	selectCard: (recipe) => dispatch(selectCard(recipe))
})

Card.propTypes = {
	addFavorite: PropTypes.func.isRequired,
	removeFavorite: PropTypes.func.isRequired,
	selectCard: PropTypes.func.isRequired,
	favorites: PropTypes.arrayOf(PropTypes.object).isRequired,
	recipe: PropTypes.shape({
		name: PropTypes.string,
		yield: PropTypes.number,
		calories: PropTypes.number,
		dietLabel: PropTypes.array,
		url: PropTypes.string,
		image: PropTypes.string,
		source: PropTypes.string,
		ingredients: PropTypes.array
	}).isRequired
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Card))
