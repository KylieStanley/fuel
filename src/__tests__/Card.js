import React from 'react'
import { shallow } from 'enzyme'
import Card from '../Card'

describe('Card', () => {
  let wrapper
  let mockRecipe

  beforeEach(() => {
  	mockRecipe = {name: 'chicken pasta', url: 'chicken.jpg'}
    wrapper = shallow(
    	<Card 
    		recipe={mockRecipe}
    	/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
