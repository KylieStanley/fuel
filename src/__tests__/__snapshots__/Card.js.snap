// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    recipe={
      Object {
        "name": "chicken pasta",
        "url": "chicken.jpg",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="image"
          style={
            Object {
              "backgroundImage": "url(undefined)",
            }
          }
        />,
        <div
          className="recipe-info"
        >
          <h3>
            chicken pasta
          </h3>
          <div
            className="card-bottom"
          >
            <button
              className="fave-btn"
            >
              <i
                className="far fa-heart"
              />
            </button>
            <p />
          </div>
        </div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "image",
          "style": Object {
            "backgroundImage": "url(undefined)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              chicken pasta
            </h3>,
            <div
              className="card-bottom"
            >
              <button
                className="fave-btn"
              >
                <i
                  className="far fa-heart"
                />
              </button>
              <p />
            </div>,
          ],
          "className": "recipe-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "chicken pasta",
            },
            "ref": null,
            "rendered": "chicken pasta",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="fave-btn"
                >
                  <i
                    className="far fa-heart"
                  />
                </button>,
                <p />,
              ],
              "className": "card-bottom",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i
                    className="far fa-heart"
                  />,
                  "className": "fave-btn",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "far fa-heart",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="image"
            style={
              Object {
                "backgroundImage": "url(undefined)",
              }
            }
          />,
          <div
            className="recipe-info"
          >
            <h3>
              chicken pasta
            </h3>
            <div
              className="card-bottom"
            >
              <button
                className="fave-btn"
              >
                <i
                  className="far fa-heart"
                />
              </button>
              <p />
            </div>
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "image",
            "style": Object {
              "backgroundImage": "url(undefined)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                chicken pasta
              </h3>,
              <div
                className="card-bottom"
              >
                <button
                  className="fave-btn"
                >
                  <i
                    className="far fa-heart"
                  />
                </button>
                <p />
              </div>,
            ],
            "className": "recipe-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "chicken pasta",
              },
              "ref": null,
              "rendered": "chicken pasta",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="fave-btn"
                  >
                    <i
                      className="far fa-heart"
                    />
                  </button>,
                  <p />,
                ],
                "className": "card-bottom",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="far fa-heart"
                    />,
                    "className": "fave-btn",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "far fa-heart",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
